McCorrections/include/PileUpWeight.h

```diff
@@ -45,5 +45,108 @@ class PileUpWeight : public IWeightProvider {
     HistPtr pu_weights;
 };
 
+class PileUpWeightEx : public IWeightProvider {
+public:
+    using Event = ntuple::Event;
+    using Hist = TH1;
+    using HistPtr = std::shared_ptr<Hist>;
+
+    PileUpWeightEx(const std::string& pu_data_file_name, const std::string& pu_mc_file_name,
+                 const std::string& cfg_file_name, double _max_available_pu,
+                 double _default_pu_weight) :
+        max_available_pu(_max_available_pu), default_pu_weight(_default_pu_weight)
+    {
+        LoadPUWeights(pu_data_file_name, pu_mc_file_name, cfg_file_name);
+    }
+
+    virtual double Get(const Event& event) const override { return GetT(event); }
+    virtual double Get(const ntuple::ExpressEvent& event) const override { return GetT(event); }
+
+    void SetActiveDataset(const std::string& active_dataset)
+    {
+        std::ostringstream ss_active_dataset;
+        ss_active_dataset << "n_pu_mc_" << active_dataset;
+        std::string active_dataset_full_name = ss_active_dataset.str();
```

Comment:
in such simple situation better to write
```c++
const std::string active_dataset_full_name = "n_pu_mc_" + active_dataset;
```

Reply:
Line `std::string active_dataset_full_name = ss_active_dataset.str();` is not needed and should not compile